core/src/sql/value/serde/ser/statement/define/user.rs

```diff
@@ -100,12 +100,76 @@ impl serde::ser::SerializeStruct for SerializeDefineUserStatement {
 			hash: self.hash,
 			code: self.code,
 			roles: self.roles,
-			session: self.session,
+			duration: self.duration,
 			comment: self.comment,
 			if_not_exists: self.if_not_exists,
 		})
 	}
 }
+pub struct SerializerDuration;
+
+impl ser::Serializer for SerializerDuration {
+	type Ok = UserDuration;
+	type Error = Error;
+
+	type SerializeSeq = Impossible<UserDuration, Error>;
+	type SerializeTuple = Impossible<UserDuration, Error>;
+	type SerializeTupleStruct = Impossible<UserDuration, Error>;
+	type SerializeTupleVariant = Impossible<UserDuration, Error>;
+	type SerializeMap = Impossible<UserDuration, Error>;
+	type SerializeStruct = SerializeDuration;
+	type SerializeStructVariant = Impossible<UserDuration, Error>;
+
+	const EXPECTED: &'static str = "a struct `UserDuration`";
+
+	#[inline]
+	fn serialize_struct(
+		self,
+		_name: &'static str,
+		_len: usize,
+	) -> Result<Self::SerializeStruct, Error> {
+		Ok(SerializeDuration::default())
+	}
+}
+
+#[derive(Default)]
+#[non_exhaustive]
+pub struct SerializeDuration {
+	pub token: Option<Duration>,
+	pub session: Option<Duration>,
+}
+
+impl serde::ser::SerializeStruct for SerializeDuration {
+	type Ok = UserDuration;
+	type Error = Error;
+
+	fn serialize_field<T>(&mut self, key: &'static str, value: &T) -> Result<(), Error>
+	where
+		T: ?Sized + Serialize,
+	{
+		match key {
+			"token" => {
+				self.token =
+					value.serialize(ser::duration::opt::Serializer.wrap())?.map(Into::into);
+			}
+			"session" => {
+				self.session =
+					value.serialize(ser::duration::opt::Serializer.wrap())?.map(Into::into);
+			}
+			key => {
+				return Err(Error::custom(format!("unexpected field `UserDuration::{key}`")));
+			}
+		}
+		Ok(())
+	}
+
+	fn end(self) -> Result<Self::Ok, Error> {
+		Ok(UserDuration {
+			token: self.token,
+			session: self.session,
+		})
+	}
+}
 
 #[cfg(test)]
 mod tests {
```

Comment:
I would probably add a second test with non default value. As often Option values are just None, which then does not test any sub structure. `Having Option<Duration>` in mind.

Reply:
You are right. Nice catch! I will add an additional test :+1:

Reply:
Done in 31d2d2df.